rfpremier/ah-django
# #164857215 Password Reset

#### What does this PR do?
- Send the reset password link to the user's email
- Create the reset password endpoint
#### Description of Task to be completed?
- Send password reset link to user email
- Create the reset password endpoint
- Write tests for reset password endpoint
#### How should this be manually tested?
* Clone the repo
```
https://github.com/rfpremier/ah-django.git
```
* Navigate to the project directory
* Create and activate a virtual environment
```
$ virtualenv env
$ source env/bin/activate
```
* Checkout to `ft/164857215-password-reset` branch
```
$ git checkout ft/164857215-password-reset
```
* Install requirements
```
$ pip install -r requirements.txt
```
* Setup your local environment
```
$ source .env
```
* Make migrations
```
$ python manage.py makemigrations
```
* Run migrations
```
$ python manage.py migrate
```
* Start the app
```
$ python manage.py runserver
```
* Open Postman 
* If you don't have an account, signup on `[POST] http://127.0.0.1:8000/api/users/`
```
{
          "user": {
                "email": "validemailaddress@example.com", # this should be a working email address that you have access to
                "username": "your username",
                "password": "yourpassword123"
            }
        }
```
* Test the reset-password on `[POST] http://127.0.0.1:8000/api/users/password-reset/` 
* Use the following payload
```
{
	"user": {
			"email": "validemailaddress@example.com" # this should be a working email address that you have access to
	}
}
```
* You will receive the following response
```
{
    "status": 200,
    "token": "token",
    "uid": id,
    "message": "Check email to reset password"
}
```
* Check your email for the reset password url
* Test the url `[PATCH] http://127.0.0.1:8000/api/users/password-reset-confirm/?uid='{uid}&token='{token}'`with postman
* Fields required
```
{
	"new_password": "newpassword123",
	"re_new_password": "newpassword123"
}
```
* You will receive the following response
```
{
    "message": "Password reset successfully.",
    "status": 200
}
```
* Run tests
```
$ python manage.py test
```
#### What are the relevant pivotal tracker stories?
#164857215 
### Screenshots (if appropriate)
![Screenshot_2019-03-28_10-11-45](https://user-images.githubusercontent.com/29132159/55139970-0ed75900-5148-11e9-9120-1ded675f7902.png)
![Screenshot_2019-03-28_10-11-10](https://user-images.githubusercontent.com/29132159/55140005-1dbe0b80-5148-11e9-9cf0-0fbc99630a92.png)
